### PR TITLE
chore: use string builder instead of string concatenation in catchup service

### PR DESCRIPTION
**Pull request by kehiy**
## Summary
hello Algorand team
I replaced string concatenation with a builder, that's faster and more efficient, concatenation with `+` is also too expensive.

### CODE REVIEW

**Comment by kehiy** [comment]
@bbroder-algo @winder can you review this?

**Review by winder — APPROVED** [review]
LGTM

**Comment by winder** [comment]
Thanks for the contribution

**Review by bbroder-algo — APPROVED** [review]
no forks

**Review by jannotti — COMMENTED** [review]
I'd rather see the whole `s` defined in one piece using backticks to avoid all the double quotes, the escaping, and concatenation altogether.

File: catchup/service.go [hunk 1]
```diff
@@ -744,17 +745,19 @@ func (s *Service) fetchRound(cert agreement.Certificate, verifier *agreement.Asy
 		if cert.Round == fetchedCert.Round &&
 			cert.Proposal.BlockDigest != fetchedCert.Proposal.BlockDigest &&
 			fetchedCert.Authenticate(*block, s.ledger, verifier) == nil {
-			s := "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n"
```
Comment (algonautshant) [inline catchup/service.go:747]: Where is the test for this change?

Comment (bbroder-algo) [reply catchup/service.go:747]: fwiw, I ran it in go playground and it worked.

**Comment by jannotti** [comment]
BenchmarkPlusEq-8     	 1515818	       779.0 ns/op
BenchmarkBuilder-8    	 3102158	       384.8 ns/op
BenchmarkConstant-8   	1000000000	         0.3593 ns/op

**Pull request merged**